### DISCUSSION TITLE
TypeError: fetch failed

### DISCUSSION THREAD

**Discussion by ubisage** [vercel/next.js#61396] [App Router]
I am having this error while creating a production build, however build is succesful . The localhost is working perfectly fine
Here is the screenshot,
![image](https://github.com/vercel/next.js/assets/105533029/43850ac6-9c0f-4116-ae23-cff02be13ed1)

**Comment by icyJoseph** [comment]
Hi,

You are most likely reaching for API Routes, or Handlers (in App Router), during build time. That won't work because there's no server to take care of them while building the app.

You also most likely have wrapped  a `try/catch` block around the fetch requests, which is very nice, and something most people forget to do.

That being said, when you run your app, you probably will not see what you expect. Not sure what kind of pages you have, pre-rendered or on-demand.

To fix this, please refer to this:

- https://nextjs-faq.com/fetch-api-in-rsc
- https://nextjs-faq.com/fetch-api-in-getserversideprops
